shared/bash_remediation_functions/dconf_settings.sh

```diff
@@ -0,0 +1,35 @@
+# Function to configure DConf settings for RHEL and Fedora systems.
+#
+# Example Call(s):
+#
+#     dconf_settings 'org/gnome/login-screen' 'banner-message-enable' 'true' 'local.d' '10-banner'
+#
+function dconf_settings {
+	local _path=$1 _key=$2 _value=$3 _db=$4 _settingFile=$5
+
+	# Check sanity of the input
+	if [ $# -ne "5" ]
+	then
+		echo "Usage: dconf_settings 'dconf_path' 'dconf_setting' 'dconf_db' 'dconf_settingsfile'"
+		echo "Aborting."
+		exit 1
+	fi
+
+	# Check for setting in any of the DConf db directories
+	SETTINGSFILES=($(grep -r "\[${_path}]" "/etc/dconf/db/" | grep -v "distro\|ibus" | cut -d":" -f1))
+	DCONFFILE="/etc/dconf/db/${_db}/${_settingFile}"
+
+	if [[ -z "${SETTINGSFILES[@]}" ]]
+	then
+		[ ! -z ${DCONFFILE} ] || $(echo "" >> ${DCONFFILE})
+		echo "[${_path}]" >> ${DCONFFILE}
+		echo "${_key}=${_value}" >> ${DCONFFILE}
+	else
+		if grep -q "${_key}" ${SETTINGSFILES[@]}
+		then
+			sed -i "s/${_key}=.*/${_key}=${_value}/g" ${SETTINGSFILES[@]}
+		else
+			sed -i "\|\[${_path}]|a\\${_key}=${_value}" ${SETTINGSFILES[@]}
```

Comment:
I have tried that and found out that there may be existing config entries that look like `key = value` (i.e. with whitespaces), that would escape the substitution.
Would you be interested in adding test for this to the SSG test suite?

Reply:
Haven't really played with code in the test suite, but sure.

Reply:
Actually, te test suite makes it possible to test remediations and testing the shared functionality alone is not supported ATM, so let's defer the testing. However, please take a look at those whitespaces.